misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt

```diff
@@ -89,6 +90,13 @@ internal class SessionFactoryService(
       applySetting(AvailableSettings.USE_GET_GENERATED_KEYS, "true")
       applySetting(AvailableSettings.USE_NEW_ID_GENERATOR_MAPPINGS, "false")
       applySetting(AvailableSettings.JDBC_TIME_ZONE, "UTC")
+      if (config.type != DataSourceType.VITESS && config.type != DataSourceType.VITESS_MYSQL) {
+        // This tells Hibernate that autocommit is always false, so Hibernate won't try to set it
+        // for every transaction.
+        // https://vladmihalcea.com/why-you-should-always-use-hibernate-connection-provider_disables_autocommit-for-resource-local-jpa-transactions/
+        // This setting doesn't seem to work with Vitess though...
```

Comment:
@tirsen Vitess tests fail with this weird error so I've disabled it.

```
Caused by: java.sql.SQLException: vtgate: http://0646a45145d0:27000/: cannot change to a non-master type in the middle of a transaction: REPLICA
```
https://app.circleci.com/pipelines/github/cashapp/misk/2747/workflows/083d539a-95da-4b87-a1c5-809e1feef318/jobs/8964